### PR TITLE
Add unit and integration test environment

### PR DESCRIPTION
**Pull request by uwefreidank**
This PR introduces a test framework for unit test and integration tests.
To prove it first unit test `binding_test` and first integration `space_controller_integration_test` have been added.
For integration test the `counterfeiter` module is used (like e.g. in https://github.com/SAP/sap-btp-service-operator)

Additional:
In order to satisfy the github.com default runner ubuntu-latest in my forked repo, the build workflow needed to be slightly adapted.

### CODE REVIEW

File: .github/workflows/build.yaml [hunk 1]
```diff
@@ -71,7 +71,7 @@ jobs:
 
   build-docker:
     name: Build Docker image
-    runs-on: ubuntu-22.04
+    runs-on: ubuntu-latest
```
Comment (TheBigElmo) [inline .github/workflows/build.yaml:74]: we only use specific versions and will update it, once the new LTS version was released and tested.

File: Meeting with Owner.md
Comment (TheBigElmo) [inline Meeting with Owner.md:1]: Please remove internal communications

**Comment by TheBigElmo** [comment]
I have updated the year for the license header. https://github.com/SAP/cf-service-operator/commit/52a5fec2c31509a45ddab40007a015fce1bfed06

**Comment by uwefreidank** [comment]
should we have a separate PR, only for updating the year in the license header of all files, as the test pipeline complains about several files.
But i don't want to mix up this with the actual test-introduction.

**Comment by uwefreidank** [comment]
Let's first merge https://github.com/SAP/cf-service-operator/pull/45 to make the build pipeline happy.

**Comment by cbarbian-sap** [comment]
Thanks for contributing. Just to understand - what is the purpose of the new `State` type and according constants? It's not used, is it?

File: Dockerfile [hunk 2]
```diff
@@ -21,6 +21,7 @@ COPY Makefile Makefile
 
 # Run tests and build
 RUN make envtest \
+ && netstat -a | grep 8080 \
```
[the remark this answers is not in the document]
Comment (uwefreidank) [reply Dockerfile]: unfortunately the build pipeline fails https://github.com/SAP/cf-service-operator/actions/runs/8435429658/job/23100858286
we need to figure out, how to satisfy the docker build step.

[the remark this answers is not in the document]
Comment (uwefreidank) [reply Dockerfile]: is there a reason, why we need to run the test within the dockerfile, @cbarbian-sap ?
the `build-docker` job needs the `test`job anyway.
seems that port 8080 is already in use, and let the envtest fail.

[the remark this answers is not in the document]
Comment (cbarbian-sap) [reply Dockerfile]: Not sure about the root cause of the bind error; either the test somehow starts up multiple controllers in parallel or does not shut them down when done (don't think so actually), or it clashes with something already running on the github runner on 8080; we usually disable metrics/health endpoints in envtest, such as here: https://github.com/SAP/redis-operator/blob/main/pkg/operator/operator_suite_test.go#L112

[the remark this answers is not in the document]
Comment (uwefreidank) [reply Dockerfile]: thxs. the metrics endpoint is now also disabled. only question, why we need a test execution during build step.

**Comment by uwefreidank** [comment]
@TheBigElmo @cbarbian-sap can we merge the PR?

**Review by TheBigElmo — CHANGES_REQUESTED** [review]
Please enable go test in the dockerfile, so it can be ensured, that the test works in this environment.

**Review by TheBigElmo — APPROVED** [review]

**Pull request merged**